### PR TITLE
update: bump up notation-core-go

### PR DESCRIPTION
**Pull request by patrickzheng200**
This PR bumps up notation-core-go. Since extended attribute key type is changed to interface{}, notation-go code needs to be updated accordingly.

Signed-off-by: Patrick Zheng <patrickzheng@microsoft.com>

### CODE REVIEW

File: verification/helpers.go [hunk 1]
```diff
@@ -69,6 +69,15 @@ func isPresent(val string, values []string) bool {
 	return false
 }
 
+func isPresentAny(val interface{}, values []interface{}) bool {
```
Comment (priteshbandi) [inline verification/helpers.go:72]: qq: can we get rid of function defined at line 63?

**Review by priteshbandi — COMMENTED** [review]
Can you please update CR with latest changes from notation-core-go

File: verification/helpers.go (hunk 1 above)
Comment (patrickzheng200) [reply verification/helpers.go:72]: > qq: can we get rid of function defined at line 63?

the function at line 63 is used in other places such as policy.go. For example, `isPresent(wildcard, statement.RegistryScopes)`.

**Review by priteshbandi — APPROVED** [review]
LGTM

File: verification/helpers.go (hunk 1 above)
Comment (shizhMSFT) [reply verification/helpers.go:72]: Can we use generics like 
```go
func isPresent[T comparable](val T, values []T) bool {
	for _, v := range values {
		if v == val {
			return true
		}
	}
	return false
}
```

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: verification/helpers.go (hunk 1 above)
Comment (shizhMSFT) [reply verification/helpers.go:72]: LoL.
```
any does not implement comparable
```

File: verification/verifier_helpers.go [hunk 2]
```diff
@@ -253,8 +253,8 @@ func (v *Verifier) executePlugin(ctx context.Context, trustPolicy *TrustPolicy,
 	if err != nil {
 		return nil, err
 	}
-	var attributesToProcess []string
-	extendedAttributes := make(map[string]interface{})
+	var attributesToProcess []interface{}
+	extendedAttributes := make(map[interface{}]interface{})
```
Comment (rgnote) [inline verification/verifier_helpers.go:257]: This can't be marshaled using `encoding/json`

**Pull request merged**